pkg/controller/kogitoinfra/mongodb.go

```diff
@@ -0,0 +1,330 @@
+			infrastructure.MongoDBAppSecretDatabaseKey:     credentials.Database,
+		},
+	}
+	if err := framework.SetOwner(i.instance, i.scheme, secret); err != nil {
+		return nil, err
+	}
+	if err := kubernetes.ResourceC(i.client).Create(secret); err != nil {
+		log.Errorf("Error occurs while creating %s", secret)
+		return nil, err
+	}
+	log.Debug("%s successfully created", secret)
+	return secret, nil
+}
+
+type mongoDBInfraReconciler struct {
+	targetContext
+}
+
+// getMongoDBWatchedObjects provide list of object that needs to be watched to maintain MongoDB kogitoInfra resource
+func getMongoDBWatchedObjects() []framework.WatchedObjects {
+	return []framework.WatchedObjects{
+		{
+			GroupVersion: mongodb.SchemeGroupVersion,
+			AddToScheme:  mongodb.SchemeBuilder.AddToScheme,
+			Objects:      []runtime.Object{&mongodb.MongoDB{}},
+		},
+		{
+			Objects: []runtime.Object{&corev1.Secret{}},
+		},
+	}
+}
+
+// Reconcile reconcile Kogito infra object
+func (i *mongoDBInfraReconciler) Reconcile() (requeue bool, resultErr error) {
+	var mongoDBInstance *mongodb.MongoDB
+
+	if !infrastructure.IsMongoDBAvailable(i.client) {
+		return false, errorForResourceAPINotFound(&i.instance.Spec.Resource)
+	}
+
+	// Step 1: check whether user has provided custom mongoDB instance reference
+	mongoDBNamespace := i.instance.Spec.Resource.Namespace
+	mongoDBName := i.instance.Spec.Resource.Name
+	if len(mongoDBNamespace) == 0 {
+		mongoDBNamespace = i.instance.Namespace
+		log.Debugf("Namespace is not provided for infrastructure MongoDB resource %s, taking instance namespace(%s) as default", i.instance.Name, mongoDBNamespace)
+	}
+	if len(mongoDBName) == 0 {
+		return false, errorForResourceConfigError(i.instance, "No resource name given")
+	}
+
+	if mongoDBInstance, resultErr = i.loadDeployedMongoDBInstance(mongoDBName, mongoDBNamespace); resultErr != nil {
+		return false, resultErr
+	} else if mongoDBInstance == nil {
+		return false, errorForResourceNotFound("MongoDB", i.instance.Spec.Resource.Name, mongoDBNamespace)
+	}
+
+	log.Debugf("Got MongoDB instance %v", mongoDBInstance)
+	if mongoDBInstance.Status.Phase != mongodb.Running {
+		return false, errorForResourceNotReadyError(fmt.Errorf("mongoDB instance %s not ready. Waiting for Status.Phase == Running", mongoDBInstance.Name))
+	}
+	log.Info("MongoDB instance is running")
+	if resultErr = i.updateMongoDBAppPropsInStatus(mongoDBInstance); resultErr != nil {
+		return false, nil
+	}
+	if resultErr = i.updateMongoDBEnvVarsInStatus(mongoDBInstance); resultErr != nil {
+		return false, nil
+	}
+	return false, resultErr
+}
+
+// MongoDBCredential holds the credentials information of a user into an mongoDB server
+type MongoDBCredential struct {
+	AuthDatabase string `yaml:"auth-database"`
+	Username     string `yaml:"username"`
+	Password     string `yaml:"password"`
+	Database     string `yaml:"database"`
+}
+
+// fillMongoDBCredential fills the credential of the MongoDB server deployed with the Kogito Operator.
```

Comment:
```suggestion
// retrieveMongoDBCredentials 
```
We might need to rethink this description.